impl11/ddraw/VRRendererOpenXR.cpp

```diff
@@ -0,0 +1,139 @@
+#include "VRRendererOpenXR.h"
+#include "EffectsCommon.h"
+#include <algorithm> // any_of
+
+using namespace std;
+
+bool g_bOpenXREnabled = false; // The user sets this flag to true to request support for OpenXR
+bool g_bUseOpenXR = false; //The systems sets this flag when an OpenXR is available and compatible
+bool g_bOpenXRInitialized = false; //The systems sets this flag when OpenXR has been initialized correctly
+
+const char* VRRendererOpenXR::ask_extensions[] = {
+	XR_KHR_D3D11_ENABLE_EXTENSION_NAME, // Use Direct3D11 for rendering
+	XR_EXT_DEBUG_UTILS_EXTENSION_NAME,  // Debug utils for extra info
+};
+std::vector<const char*> VRRendererOpenXR::use_extensions;
+
+bool VRRendererOpenXR::is_available()
+{
+	// OpenXR will fail to initialize if we ask for an extension that OpenXR
+	// can't provide! So we need to check our all extensions before 
+	// initializing OpenXR with them. Note that even if the extension is 
+	// present, it's still possible you may not be able to use it. For 
+	// example: the hand tracking extension may be present, but the hand
+	// sensor might not be plugged in or turned on. There are often 
+	// additional checks that should be made before using certain features!
+
+
+	// We'll get a list of extensions that OpenXR provides using this 
+	// enumerate pattern. OpenXR often uses a two-call enumeration pattern 
+	// where the first call will tell you how much memory to allocate, and
+	// the second call will provide you with the actual data!
+	uint32_t ext_count = 0;
+	XrResult xrResult = XR_ERROR_RUNTIME_FAILURE;
+	xrResult = xrEnumerateInstanceExtensionProperties(nullptr, 0, &ext_count, nullptr);
+	vector<XrExtensionProperties> xr_exts(ext_count, { XR_TYPE_EXTENSION_PROPERTIES });
+	xrResult = xrEnumerateInstanceExtensionProperties(nullptr, ext_count, &ext_count, xr_exts.data());
+
+	if (xrResult != XR_SUCCESS)
+		return false;
+
+	log_debug("[DBG][OpenXR] Extensions available");
+	for (size_t i = 0; i < xr_exts.size(); i++) {
+		log_debug("- %s\n", xr_exts[i].extensionName);
```

Comment:
I think you need to add the "[DBG]" in this log_debug call

Reply:
Indeed. Fixed.